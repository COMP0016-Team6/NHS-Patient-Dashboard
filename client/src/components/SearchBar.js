import React, {useState, useEffect} from "react";
import {Link} from "react-router-dom";

// each patient contains user_id, user_name, user_email
const SearchBar = ({ patients, select, setSelectedPatients }) => {
  const [searchTerm, setSearchTerm] = useState("");
  const [searchResults, setSearchResults] = useState([]);
  const onChange = e => {
    setSearchTerm(e.target.value);
  }

  useEffect(() => {
    const results = patients.filter(patient => patient.user_name.toLowerCase().includes(searchTerm.toLowerCase()));
    setSearchResults(results);
  }, [searchTerm]);

  return (
    <>
      <input
        type="text"
        placeholder="Search By Name"
        value={searchTerm}
        onChange={onChange}      
      />
      <div>
        {searchTerm === ""?
          patients.map(patient => (
            <h3 key={patient.user_id} className="text-info">
            {typeof(select)==="undefined"? null : <input type="checkbox"/>}
            <Link to={`/dashboard/${patient.user_id}`}>{patient.user_name}</Link>
            </h3>
          )):
          searchResults.map(patient => (
            <h3 key={patient.user_id} className="text-info">
              {typeof(select)==="undefined"? null : <input type="checkbox"/>}
              <Link to={`/dashboard/${patient.user_id}`}>{patient.user_name}</Link>
            </h3>
          ))
        }
      </div>
    </>
  )
}

export default SearchBar;