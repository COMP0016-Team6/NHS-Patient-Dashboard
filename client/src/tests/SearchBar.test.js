import React from "react";
import SearchBar from "../components/SearchBar";
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { clinicianWrapper } from "./util";

it ("SearchBar snapshot matches", () => {
  const tree = render(clinicianWrapper(<BrowserRouter><SearchBar /></BrowserRouter>));
  expect(tree).toMatchSnapshot();
});
