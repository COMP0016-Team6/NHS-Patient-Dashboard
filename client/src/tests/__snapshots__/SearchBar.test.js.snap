// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar snapshot matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root form-control my-3"
        style="color: rgb(97, 218, 251);"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          Search By Name
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="mt-4"
      >
        <h3>
          <a
            href="/dashboard/2"
          >
            Test User Two
          </a>
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiTextField-root form-control my-3"
      style="color: rgb(97, 218, 251);"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
      >
        Search By Name
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="mt-4"
    >
      <h3>
        <a
          href="/dashboard/2"
        >
          Test User Two
        </a>
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
